gateway/internal/types/types.go

```diff
@@ -27,6 +27,8 @@ type AuthRequestContext struct {
 	SourceJobRunID  string
 	SourceTaskRunID string
 	Source          backendconfig.SourceT
+	// DestinationID is optional param, destination id will be present for rETL Request
```

Comment:
Not sure if we need this comment. `SourceJobRunID` & `SourceTaskRunID` are both optional as well & we don't mention when they are going to be present. One reason why we would choose not to include such comments would be change, i.e. at some point we will decide to change (when these will be present) and almost certainly we'll forget to update the relevant comments! So we end up with invalid comments!

Reply:
IMO its better to keep this comment, and if we start using this field for cases we should update the comment, this way we can keep more readable and intuitive. Or we can just add a comment that this field is optional. That way reading through the code might give us a better understanding that these fields may or many not necessarily 
always be available.